src/main/java/com/cloudbees/jenkins/plugins/advisor/AdvisorGlobalConfiguration.java

```diff
@@ -211,8 +211,8 @@ public void setEmail(@CheckForNull String email) {
     return email;
   }
 
-  public @Nonnull String getPassword() {
-    return Secret.toString(password);
+  public @Nonnull Secret getPassword() {
```

Comment:
My expectation here, and I admit I haven't actually tried, is that <f:password> in the index.jelly will handle `Secret` correctly and "do the right thing", which is to put a scrambled value in the field that round trips.